src/yardstick/tool/syft.py

```diff
@@ -110,23 +126,105 @@ def _install_from_git(
         abspath = os.path.abspath(path)
         if not tool_exists:
             logging.debug(f"installing syft to {abspath!r}")
-            # pylint: disable=line-too-long
-            c = f"go build -ldflags \"-w -s -extldflags '-static' -X github.com/anchore/syft/internal/version.version={description}\" -o {abspath} ./cmd/syft/"
-            logging.debug(f"running {c!r}")
-            subprocess.check_call(
-                shlex.split(c),
-                stdout=sys.stdout,
-                stderr=sys.stderr,
-                cwd=repo_path,
-                env=dict(**{"GOBIN": abspath, "CGO_ENABLED": "0"}, **os.environ),
-            )
-
-            os.chmod(f"{path}/syft", 0o755)
+            cls._run_go_build(abs_install_dir=abspath, repo_path=repo_path, description=description, binpath=path)
         else:
             logging.debug(f"using existing syft installation {abspath!r}")
 
         return Syft(path=path, version_detail=description)
 
+    @classmethod
+    def _local_build_version_suffix(cls, src_path: str) -> str:
```

Comment:
Do you think it's worth putting this in a common place? It's the same for any path that's a git repo.

Reply:
yea - I think I can extract` _hash_file` to be static and away from the class structure and then have this as a common standalone method

Reply:
fixed with [c008487](https://github.com/anchore/yardstick/pull/115/commits/c008487560b5ff722d71ef1359c67c57d66434d2)